src/index.js

```diff
@@ -1,12 +1,51 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import './index.css';
-import App from './App';
-import * as serviceWorker from './serviceWorker';
-
-ReactDOM.render(<App />, document.getElementById('root'));
-
-// If you want your app to work offline and load faster, you can change
-// unregister() to register() below. Note this comes with some pitfalls.
-// Learn more about service workers: https://bit.ly/CRA-PWA
-serviceWorker.unregister();
+import React, { useState, useRef, useEffect } from "react";
+import ReactDOM from "react-dom";
+import { PlayCircleTwoTone } from '@ant-design/icons';
+import "./styles.css";
+
+function App() {
+  const audioPlayer = useRef(null);
+  const [progress, setProgress] = useState(80);
+
+  useEffect(() => {
+    console.log(progress, audioPlayer.current.duration);
+  }, [progress]);
+  const [isPlay, setPlay] = useState(true);
+  const toggleAudio = () => {
+    if (isPlay) {
+      audioPlayer.current.play();
+      setPlay(false);
+    } else {
+      audioPlayer.current.pause();
+      setPlay(true);
+    }
+  };
```

Comment:
¿Puede ser que la lógica esté invertida? `isPlay` diría si está reproduciéndose el audio o no, ¿verdad? De ser así, quizá convenga:
```suggestion
  const toggleAudio = () => {
    if (isPlaying) {
      audioPlayer.current.pause();
      setIsPlaying(false);
    } else {
      audioPlayer.current.play();
      setIsPlaying(true);
    }
  };
```